cpp/src/reductions/reductions.cu

```diff
@@ -168,24 +31,29 @@ gdf_scalar reduction(const gdf_column *col,
 
     switch(op){
     case GDF_REDUCTION_SUM:
-        cudf::type_dispatcher(col->dtype,
-            ReduceDispatcher<cudf::reductions::ReductionSum>(), col, &scalar);
+        cudf::reductions::reduction_sum(col, &scalar);
```

Comment:
Since `scalar` is an output, can the `reduction_*` functions be made to _return_ `scalar` instead of using an output parameter? To get the output `dtype`, just pass it as an input parameter.

Reply:
`gdf_scalar` must be constructed before `cudf::reductions::xxx` calls since it must returns invalid scalar if col->size <= col->null_count.
https://github.com/rapidsai/cudf/blob/d1ba1e8b52df5f16d023dcb727468ba205443071/cpp/src/reductions/reductions.cu#L24-L31

 `cudf::reductions::xxx` should have a parameter of `gdf_scalar` since it is already constructed.
We can add additional parameter `output_dtype`, but I'd like to avoid call overhead for additional parameter.